parsl/tests/sites/test_concurrent.py

```diff
@@ -0,0 +1,48 @@
+"""Tests of the interfaces to Python's concurrent library"""
+
+from pytest import mark, warns
+
+from parsl import Config, HighThroughputExecutor
+from parsl.concurrent import ParslPoolExecutor
+
+
+def f(x):
+    return x + 1
+
+
+# TODO (wardlt): For reasons I don't understand, re-using a configuration is a problem. This creates a fresh one
+def make_config(run_dir):
```

Comment:
in the rest of the test suite, this is called `fresh_config`

Reply:
> # TODO (wardlt): For reasons I don't understand, re-using a configuration is a problem. This creates a fresh one

A Config() object in parsl is not a description of the components (such as executors and launchers) that will be used in a run. It *is* the components that will be used.

So, for example, a config might contain a high throughput executor, and that high throughput executor will be configured with script directories etc, started up, used and then shut down. None of the parsl code is built on the basis that after shutting various components down, you can reconfigure them and start them up again - which is what happens when you re-use a config.

The *description* of the configuration, which *can* be re-used, is the code that generates a Config() object - and the way to re-use that is to put it in a function, so that you can generate a fresh config object when you need to start a new DFK.

In that sense, the "configuration file" equivalent is not the Config object itself, but the source code that generates a config object.